core/eth/validator_quorum_lookup.go

```diff
@@ -0,0 +1,124 @@
+package eth
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/big"
+
+	regcoordinator "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDARegistryCoordinator"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+// A utility for looking up which quorums a given validator is a member of at a specific reference block number.
+type ValidatorQuorumLookup interface {
+	// Get the list of quorums that the given validator is a member of, at the specified reference block number.
+	GetQuorumsForValidator(
+		ctx context.Context,
+		validatorAddress core.OperatorID,
+		referenceBlockNumber uint64) ([]core.QuorumID, error)
+}
+
+var _ ValidatorQuorumLookup = (*validatorQuorumLookup)(nil)
+
+// A standard implementation of the ValidatorQuorumLookup interface.
+type validatorQuorumLookup struct {
+	registryCoordinator *regcoordinator.ContractEigenDARegistryCoordinator
+}
+
+// Create a new ValidatorQuorumLookup instance.
+func NewValidatorQuorumLookup(
```

Comment:
Did you intentionally not implement this constructor?

Reply:
Not intentional, this was an oversight. Now fixed.